### ISSUE TITLE
CI fails when SQL Server 2022 CUs can't be dowloaded

### ISSUE THREAD

**Issue by arthurschreiber** [tediousjs/tedious#1666]
@dhensby Every now and then I see the warning `Warning: Unable to download cumulative updates` from `tediousjs/setup-sqlserver@v2`, and if that happens, builds on SQL Server 2022 fail because the TDS 8 feature is broken on an unpatched 2022 version.

Can you take a look into why downloading the updates fails sporadically?

**Comment by dhensby** [comment]
This is the line where the error comes from: https://github.com/tediousjs/setup-sqlserver/blob/v2.1.0/src/utils.ts#L199

There's not much logging going on here, so it's hard to know exactly why this is happening.

It could be:

1. We have a bad status code (ie: error response from the server)
2. The regex for pulling out the install link is not working.

I'll add a PR to increase logging so we can see what is going on here.

**Comment by dhensby** [comment]
Something I had thought about when first authoring the cumulative updates feature was using the CI cache to hold onto these update URLs to prevent having to repeatedly hit the website for them, but the caching is more for caching files and not arbitrary data (https://github.com/actions/toolkit/tree/main/packages/cache).

**Comment by arthurschreiber** [comment]
Yeah. It would be nice to cache all the installer files, but I guess that can easily bump into the 10GB actions cache limit.

**Comment by dhensby** [comment]
Do the URLs change regularly, or could the action ship with some defaults, which could then be used as a fallback if the check doesn't work?

**Comment by dhensby** [comment]
@arthurschreiber has this issue happened again recently?
